src/sentinel.c

```diff
@@ -1516,6 +1536,22 @@ void rewriteConfigSentinelOption(struct rewriteConfigState *state) {
             rewriteConfigRewriteLine(state,"sentinel",line,1);
         }
 
+        /* sentinel rename-config */
+        if (master->config_command != SENTINEL_CONFIG_COMMAND) {
```

Comment:
I believe this check to be accurate. The only time we need to write to the config file is when the command differs from the default.